ci-operator/config/open-telemetry/opentelemetry-operator/open-telemetry-opentelemetry-operator-main.yaml

```diff
@@ -0,0 +1,112 @@
+base_images:
+  cli:
+    name: "4.14"
+    namespace: ocp
+    tag: cli
+  operator-sdk:
+    name: "4.14"
+    namespace: origin
+    tag: operator-sdk
+  os:
+    name: ubi
+    namespace: ocp
+    tag: "8"
+binary_build_commands: make build
+build_root:
+  image_stream_tag:
+    name: builder
+    namespace: ocp
+    tag: rhel-8-golang-1.20-openshift-4.14
+images:
```

Comment:
The E2E tests for the target allocator are failing because the target allocator image is not being built.

```sh
    logger.go:42: 09:03:57 | targetallocator-features/0-install | Error from server (BadRequest): container "ta-container" in pod "stateful-targetallocator-5c5f8cc849-6rjmd" is waiting to start: trying and failing to pull image
```

Folder where the `Dockerfile` is hosted: https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator

Also, you need to replace the name of the target allocator image by the one set in the KUTTL files (https://github.com/open-telemetry/opentelemetry-operator/blob/3055a7eae55953f55f5c9b141fd4da6d71130478/Makefile#L200C1-L201C90)

Reply:
@iblancasa  BTW can we build the target allocator in the upstream and change the image ref for it in the test files so that we do not have to do any hack.

Reply:
Scratch that previous thought, I can get this working similar to the asserts-e2e image. 
https://github.com/openshift/release/pull/40442/files#diff-a7de146804f986df22a9cc28c0037e7535212ab6037f1b54172f6113b5ca8523R18
https://github.com/openshift/release/pull/40442/files#diff-a5651e6d49373db34bcecb25cb83769aff18a2ecb0eac2b88cf498c16256e2f2R16

Anyways, will cover this change in a different PR.